### DISCUSSION TITLE
Flow-IPC for Interprozessor-Communication on a Soc?

### DISCUSSION THREAD

**Discussion by burgerTo** [Flow-IPC/ipc#116] [Ideas]
Hi,

While researching a good abstraction for IPC, I came across this interesting project. Our use case involves interprocessor communication on a SoC (System on a Chip). Specifically, it concerns communication between a Linux system (ARM cores) and a bare-metal/freeRTOS application on R5 real-time cores. Although the scenario seems different at first glance, there are some similarities.

There is a “pipe-like” bidirectional connection between the cores based on rpmsg, but it is not very performant. Additionally, both cores can access shared memory.

Is it possible to implement corresponding abstractions in flow-ipc?

- unstructured channel for rpmsg
- structured channel for shm across processor boundaries

A bare-metal/freeRTOS port of flow-ipc is not considered here. This would be a separate topic.

Best regards
Toni

**Comment by ygoldfeld** [comment]
Toni - @burgerTo - my deepest apologies in the delay in responding. Simply, I missed it since returning from paternity leave and getting into heavy development again (which continues).

If you still have interest in this topic, we should perhaps chat one-on-one first - for efficiency really. Ping me please at ygoldfel@akamai.com - we can go from there.
